app/uk/gov/hmrc/gform/sharedmodel/formtemplate/FormatExpr.scala

```diff
@@ -160,7 +160,81 @@ object RoundingMode {
   )
 }
 
-sealed trait TextConstraint
+sealed trait TextConstraint {
+  val defaultSize: Int = this match {
```

Comment:
Can we make this more readable like this:

```
sealed trait TextConstraint {
  val defaultSize: Int = this match {
    case Number(maxWholeDigits, maxFractionalDigits, _, _) =>
      if (maxFractionalDigits > 0)
        maxWholeDigits + maxFractionalDigits + 1
      else
        maxWholeDigits
    case PositiveNumber(maxWholeDigits, maxFractionalDigits, _, _) =>
      if (maxFractionalDigits > 0)
        maxWholeDigits + maxFractionalDigits + 1
      else
        maxWholeDigits
    case BasicText                    => TextConstraint.defaultSizeBasicText
    case ShortText(_, max)            => max
    case Lookup(_)                    => TextConstraint.defaultSizeLookup
    case TextWithRestrictions(_, max) => max
    case Sterling(_, _)               => TextConstraint.defaultSizeSterling
    case UkBankAccountNumber          => TextConstraint.defaultSizeUkBankAccountNumber
    case UkSortCodeFormat             => TextConstraint.defaultSizeUkSortCodeFormat
    case SubmissionRefFormat          => TextConstraint.defaultSizeSubmissionRefFormat
    case TelephoneNumber              => TextConstraint.defaultSizeTelephoneNumber
    case Email                        => TextConstraint.defaultSizeEmail
    case EmailVerifiedBy(_, _)        => TextConstraint.defaultSizeEmailVerifiedBy
    case UTR                          => TextConstraint.defaultSizeUTR
    case NINO                         => TextConstraint.defaultSizeNINO
    case UkVrn                        => TextConstraint.defaultSizeUkVrn
    case CountryCode                  => TextConstraint.defaultSizeCountryCode
    case NonUkCountryCode             => TextConstraint.defaultSizeNonUkCountryCode
    case CompanyRegistrationNumber    => TextConstraint.defaultSizeCompanyRegistrationNumber
    case EORI                         => TextConstraint.defaultSizeEORI
    case UkEORI                       => TextConstraint.defaultSizeUkEORI
    case ChildBenefitNumber           => TextConstraint.defaultSizeChildBenefitNumber
  }
}
```

Reply:
Agreed.